src/Service/NuRaftStateMachine.cpp

```diff
@@ -462,8 +462,9 @@ void NuRaftStateMachine::create_snapshot(snapshot & s, async_result<bool>::handl
         auto t2 = Poco::Timestamp().epochMicroseconds();
         auto snap_copy = snapshot::deserialize(*snp_buf);
         auto t3 = Poco::Timestamp().epochMicroseconds();
-        snap_task = std::make_shared<SnapTask>(snap_copy, store.zxid, store.session_id_counter, when_done);
+        snap_task = std::make_shared<SnapTask>(snap_copy, store, when_done);
         auto t4 = Poco::Timestamp().epochMicroseconds();
+        snap_task_ready = true;
         LOG_INFO(log, "Async create snapshot time cost {}us, {}us, {}us", (t2 - t1), (t3 - t2), (t4 - t3));
```

Comment:
1. Just remove t2/t3 for they are only for testing.
2. LOG_INFO(log, "Scheduling asyncronously creating snapshot task,  time cost {}us", (t2 - t1));